pandas/tests/arithmetic/test_numeric.py

```diff
@@ -959,77 +959,68 @@ def test_frame_operators(self, float_frame):
                 assert (df + df).equals(df)
                 tm.assert_frame_equal(df + df, df)
 
-    # TODO: taken from tests.series.test_operators; needs cleanup
-    def test_series_operators(self):
-        def _check_op(series, other, op, pos_only=False):
-            left = np.abs(series) if pos_only else series
-            right = np.abs(other) if pos_only else other
-
-            cython_or_numpy = op(left, right)
-            python = left.combine(right, op)
-            if isinstance(other, Series) and not other.index.equals(series.index):
-                python.index = python.index._with_freq(None)
-            tm.assert_series_equal(cython_or_numpy, python)
-
-        def check(series, other):
-            simple_ops = ["add", "sub", "mul", "truediv", "floordiv", "mod"]
-
-            for opname in simple_ops:
-                _check_op(series, other, getattr(operator, opname))
-
-            _check_op(series, other, operator.pow, pos_only=True)
-
-            _check_op(series, other, ops.radd)
-            _check_op(series, other, ops.rsub)
-            _check_op(series, other, ops.rtruediv)
-            _check_op(series, other, ops.rfloordiv)
-            _check_op(series, other, ops.rmul)
-            _check_op(series, other, ops.rpow, pos_only=True)
-            _check_op(series, other, ops.rmod)
+    @pytest.mark.parametrize(
+        "func",
+        [lambda x: x * 2, lambda x: x[::2], lambda x: 5],
+        ids=["multiply", "slice", "constant"],
+    )
+    def test_series_operators_arithmetic(self, all_arithmetic_functions, func):
+        op = all_arithmetic_functions
+        series = tm.makeTimeSeries().rename("ts")
+        other = func(series)
+        left = np.abs(series) if op in (ops.rpow, operator.pow) else series
+        right = np.abs(other) if op in (ops.rpow, operator.pow) else other
+
+        cython_or_numpy = op(left, right)
```

Comment:
might be worth making L974-978 a helper in either tm or tests.arithmetic.common.